tools/attach/handlers/page/ajaxupload.php

```diff
@@ -8,7 +8,10 @@
 $hasTempTag = (isset($_GET['tempTag'])
     && preg_match("/^{$this->config['temp_tag_for_entry_creation']}_[A-Fa-f0-9]+$/m", $_GET['tempTag']));
 
-if ($this->HasAccess('write') || ($this->HasAccess('read') && $hasTempTag)) {
+if (
+    ($this->LoadPage($this->getPageTag()) && $this->HasAccess('write'))
```

Comment:
@mrflos je me rends compte que j'ai laissé passer un truc.
En utilisant `$this->LoadPage($this->getPageTag())`, j'ai l'impression que ça empêche d'ajouter des fichiers ou des images lors de la création d'une page alors que c'est possible, une fois la page créée.
Quelle est la raison qui a fait qu'il t'a semblé nécessaire de rajouter ce test ?
_J'ai l'impression que les autres correctifs empêchent l'utilisation non recherchée de fichiers et que ce test n'est donc plus nécessaire. J'ai donc besoin d'éclairage._

Reply:
si tu regardes le rapport de vulnérabilité, c'est l'une des possibilités pour facilement uploader un script malicieux, en prennant une url minimaliste, car qu'importe le nom de la page, existante ou pas,ca passe.
Je pensais que pour les pages non existantes, il fallait passer par ton mécanisme de generation de tags temporaire, ce qui me parraissait bien car il faut ouvrir en mode edition pour obtenir en javascript ce complement, pour apres sauver.
Ca ne marche pas comme ca?

Reply:
Le tag temporaire n'est actuellement utilisé que pour `textelong` en mode wiki lors de l'ajout d'un fichier.
Il n'est pas utilisé pour les pages car le nom de la page est déjà connu (vu qu'il est dans l'url).
Quoiqu'il en soit, si on veut laisser la possibilité à une personne bien intentionnée d'envoyer des fichiers dès la création de la page, alors il sera toujours possible à une personne mal-intentionnée d'envoyer des fichiers sur le serveur juste avec la bonne url (quitte à d'abord charger la page d'édition pour récupérer un code anti crsf).
Du coup, je ne pense pas que ce soit une bonne idée d'empêcher l'ajout de fichiers sauvages et qu'il nous fait vraiment être attentif sur l'usage des `include` à partir d'une information non fiable (`$_GET`ou `$_POST`) pour être sûr que le script malicieux n'est pas utilisable (genre en bloquant l'ajout de ficher php et en empêchant d'appeler des templates qui sont dans files.

Reply:
je serai pour implémenter le mécanisme similaire a textelong pour les pages wikis vides, comme ca c'est la même logique de partout, et on pourra durcir/changer le mécanisme de partout plus tard.
Apres c'est sur que tant qu'il y a des pages ouvertes au public en édition, ya des risques..